TempSensor.ino

```diff
@@ -210,51 +292,73 @@ class MovingAverage {
       m_sum = m_sum + value;
       m_rotating_counter.increment();
     }
+    void reset(){
+      m_rotating_counter.resetCounter();
+    }
 };
 
 void setup() {
+  // set up trigger button
+  trigger.begin();
+  // set up laser
+  laser.begin();
+  // set up IR sensor
   mlx.begin();
   // set up the LCD's number of columns and rows:
   lcd.begin(16, 2); 
-  lcd.home ();
+  lcd.home();
 }
 
 
 MovingAverage avg_celsius;
 MovingAverage avg_fahrenheit;
 
 void loop() {
-  // Turn off the display:
-  lcd.noDisplay();
-  delay(refreshTempReading);
-
-  // Get temperature readings
-  int temp_celcius = mlx.readObjectTempC();
-  int temp_fahrenheit = mlx.readObjectTempF();
-
-  // Populate temperature values for moving average
-  if (flagMovingAverageEnable == true) {
-    avg_celsius.pushValue(temp_celcius);
-    avg_fahrenheit.pushValue(temp_fahrenheit);
-  }
+  int temp_celcius;
+  int temp_fahrenheit;
+
+  switch (trigger.readTriggerStatus()){
+    case LOW:
+      // Start laser
+      laser.on();
+
+      // Get temperature readings
+      temp_celcius = mlx.readObjectTempC();
+      temp_fahrenheit = mlx.readObjectTempF();
+      // Populate temperature values for moving average
+      if (FLAG_MOVING_AVG_ENABLE == true) {
+        avg_celsius.pushValue(temp_celcius);
```

Comment:
Not necessary for now but you can probably have just 1 'if(FLAG_MOVING_AVG_ENABLE)' and have the code for displaying data in C and F in the same place. 
In other words, having a separate check for MOVING_AVG_ENABLE for C and F is not required